Classes/UI/SparkConnectingProgressViewController.m

```diff
@@ -449,7 +449,7 @@ -(void)getDeviceAndFinishSetup
                 self.setupResult = SparkSetupMainControllerResultSuccess;
             }
             
-            dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, 1 * NSEC_PER_SEC);
```

Comment:
I am ok with this, given that you tested it.  but i would rather put energy into  solution that blocks longer and looks good. and tolerates a skip of the waiting, after we confirm things are good and ready.   layer 1 is that google is reached.  layer 2 is that we hit a route on our cloud and confim hub is online.  potentially:   cloud check for any device report's presence in the last 5 mins.  We may need a new route.

Reply:
Should I change this to 2 then?